### PR TITLE
WT-12608 Remove the skipping tree logic

### PR DESCRIPTION
**Pull request by Siddhartha8899**
In this PR, I have eliminated the skipping tree logic. I will be running perf tests aggressively before merging this change.

If there happens to be a regression, it would be better to find alternative ways of fixing it.

- I have run WT perf 5 times and see no regressions. 
- I will be running mongo perf soon.

### CODE REVIEW

File: src/evict/evict_lru.c [hunk 1]
```diff
@@ -1555,15 +1555,6 @@ __evict_walk(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue)
             continue;
         }
 
-        /*
-         * If we are filling the queue, skip files that haven't been useful in the past.
-         */
-        if (btree->evict_walk_period != 0 && btree->evict_walk_skips++ < btree->evict_walk_period) {
-            WT_STAT_CONN_INCR(session, cache_eviction_server_skip_trees_not_useful_before);
```
Comment (kommiharibabu) [inline src/evict/evict_lru.c:1562]: As the stat is not used any more, remove the stat also.

Comment (Siddhartha8899) [reply src/evict/evict_lru.c:1562]: Nice catch - updated in [c3679c2](https://github.com/wiredtiger/wiredtiger/pull/10283/commits/c3679c22cd1df8d6ae905dccea97c50c53834726)

**Comment by Siddhartha8899** [comment]
While updating [f7ed75e](https://github.com/wiredtiger/wiredtiger/pull/10283/commits/f7ed75e22ff6f12d1d39d835b4f6ea8825aa08db) commit in the PR, I realized the `urgent_queued` bool in `__evict_walk_tree` is unnecessary and removed it. However, I identified an issue where `cache_eviction_pages_queued_oldest`  and `cache_eviction_pages_queued_urgent_hs_dirty` stat is updated regardless of successful queuing. An unsuccessful case of queueing to an urgent queue could be If the urgent queue is full, however, in that case it seems better to move the page to the regular queue instead of skipping it (present behaviour).

Thoughts? If yes, I will create a seperate ticket for this change.

**Review by quchenhao — COMMENTED** [review]
Can we run wt perf tests, full mongodb tests and sys-perf test on this?

**Review by agorrod — COMMENTED** [review]
@Siddhartha8899  - I'm OOO for a few weeks now.

If you want to make progress while I'm away, please touch base with @keitharnoldsmith about the changes, otherwise I'm happy to pick it up when I'm back online.

**Pull request closed without merging**